### PR TITLE
Artifact abstraction added

### PR DESCRIPTION
**Pull request by nerlichman**
An abstraction of `RuntimeAction` is created:  `RuntimeArtifactAction`

This abstraction is an idea that came up in a discussion with @gdaniel looking to solve this issue: https://github.com/xatkit-bot-platform/xatkit/issues/225

### CODE REVIEW

**Comment by gdaniel** [comment]
Hi @nerlichman, thanks for the PR!
I see a couple of things that could be improved before merging it to master:
- There is a lot of duplicated code between `RuntimeArtifactAction` and `RuntimeMessageAction`, and both inherit from `RuntimeAction`. To me it looks like `RuntimeMessageAction` should be a subclass of `RuntimeArtifactAction`, conceptually a message is a particular kind of artifact. This would allow to keep all the duplicated code in `RuntimeArtifactAction` and only put message-related stuffs in `RuntimeMessageAction`
- The `compute()` method in `PostArtifact` is the same as the one in `PostAttachmentsMessage`, but not the same as `PostFileMessage`, which is normal. This structure is a bit confusing, maybe there is a reason for it I didn't get, but to me it looks like we don't need to provide a `compute()` implementation in `PostArtifact`. In fact, if we remove it from `PostArtifact` the class itself only contains the `channel` information, and I suggest to remove it entirely and make `PostFileMessage` and `PostAttachmentsMessage` directly inherit from `RuntimeArtifactAction`. This implies to move the `channel` information to these classes, but since we can't have multiple inheritance and Slack actions inherit both from `RuntimeMessageAction` and `RuntimeArtifactAction` I dont' think we have a better solution without changing too many things.

In summary, I'd refactor `RuntimeArtifactAction` and `RuntimeMessageAction` to make the latter inherit from the former, and factorize duplicated code. Then I'd probably remove `PostArtifact` from the Slack plugin and make the actions directly inherit from `RuntimeMessageAction` and `RuntimeArtifactAction` when needed.

Oh and don't forget to remove these comments from the code :)
```
/**
 * TODO: add method to {@link RuntimeMessageAction} to allow being called
 * without defining a message parameter
 */
```

What do you think?

**Comment by nerlichman** [comment]
@gdaniel I've just pushed the changes you've mentioned.

Please let me know if think other changes could be done/are needed.
